### ISSUE TITLE
further ideas to make it perfect

### ISSUE THREAD

**Issue by JR-Utily** [aviggiano/guitar-pro-6#2]
Hi,

The work you did was a big help for me to be able to run Guitar Pro on newer distro without having to think too much. With the end of the linux support by Arobas, this is really great.

Here are some things I added :
- To be able to run it without typing my credencials everytime, I copied files from `/root/.config`, `/root/.ArobasMusic` and the `/opt/GuitarPro6/GuitarPro.dat` outside of the container (with `docker cp`) and volume them in the run
- To be able to have sound with pulseaudio, I added `ENV PULSE_SERVER /run/pulse/native` in the Dockerfile and volume my /run/user/1000/pulse to /run/pulse in the run
- To use all Banksound I own, I did a backup of Data folder from a former GP install and volume it in the run
- To simplify the execution, I added `CMD /opt/GuitarPro6/launcher.sh ` in the Dockerfile
- To be able to access my gpx files, I volume my dropbox folder in the run

In the end, here is my Dockerfile and my run command:

```
FROM aviggiano/guitar-pro-6
ENV PULSE_SERVER /run/pulse/native
CMD /opt/GuitarPro6/launcher.sh
```

```
#!/usr/bin/env bash

xhost +local:root

GP_DATA_PATH="$1"
DROPBOX_PATH="$2"
GP_USER_PATH="$3"

docker run --rm \
	--name gppro \
	-e DISPLAY -v /tmp/.X11-unix:/tmp/.X11-unix/ \
	--device /dev/snd \
	-v /run/user/1000/pulse:/run/pulse:ro \
	-v "$GP_DATA_PATH":/opt/GuitarPro6/Data:ro \
	-v "$DROPBOX_PATH":/root/dropbox \
	-v "$GP_USER_PATH"/home.config:/root/.config \
	-v "$GP_USER_PATH"/home.ArobasMusic:/root/.ArobasMusic \
	-v "$GP_USER_PATH"/opt.GuitarPro.dat:/opt/GuitarPro6/GuitarPro.dat \
	my-guitar-pro
```

The trickier part was for the credential. Now I've done it once and for all, but I wonder how it could be automatized for new users (as you need to run the container, fill in your credential, then outside the container run some `docker cp`, things are done in 2 different scopes... )

Thank you again for the good work (so no issue here, just sharing my recipes :) )

**Comment by aviggiano** [comment]
Hey, thank you for your comment! This is exactly why I decided to put this on github. After all, I am just a music/docker enthusiast who wanted to use Guitar Pro on Linux.

Would you be so kind to submit a pull request with the changes you made to the dockerfile? And maybe to the README, explaining what your run command does and why you need these config variables.

By the way, I haven't really used this project in a while, since I haven't had time to transcribe my music pieces. Is it still working? Did you encounter any bugs? Do you have any clue on how to debug #1 ?

Thanks!

**Comment by JR-Utily** [comment]
Ok, I'll try to do something this week end if I could find time.
This work pretty well, I've just used it on a computer where I can't install GP anymore, and with my modifications, everything was like a charm. 

The only thing I'm afraid is when the repos from 16.04 aren't support anymore, it will become difficult to maintain this (I'm not a fan of wgetting packages one by one instead of simply running apt get). For now it is ok, but in a few months/years it will become painfull.

**Comment by aviggiano** [comment]
Closed by #3
